core/src/main/java/org/apache/accumulo/core/spi/balancer/TableLoadBalancer.java

```diff
@@ -188,6 +188,22 @@ public void getAssignments(AssignmentParameters params) {
     }
   }
 
+  @Override
+  public boolean needsReassignment(CurrentAssignment currentAssignment) {
+    var tableId = currentAssignment.getTablet().getTable();
+    String value = environment.getConfiguration(tableId).get(TABLE_ASSIGNMENT_GROUP_PROPERTY);
+    String expectedGroup = (value == null || StringUtils.isEmpty(value))
+        ? Constants.DEFAULT_RESOURCE_GROUP_NAME : value;
+
+    if (!expectedGroup.equals(currentAssignment.getResourceGroup())) {
```

Comment:
Pushing the resource group comparison inside the plugin makes the interaction between Accumulo and plugin more generic.  This allows different impls, like the host reg ex balancer could possibly implement this method.